### PR TITLE
Update events logic to allow mapping specific events to specific tags

### PR DESCRIPTION
**Pull request by jsun-m**
HTTP Inputs in Fluent-bit dynamically set `tags` based off the URI of the request.
e.g. a request to `fluentbit.monitoring:9880/tag_name` results in fluent-bit automatically setting a tag of `tag_name` to the event

These tag names are then matched with an output in the example below
```
[OUTPUT]
          Name  http
          Match tag_name
          ....
```

By default, the events will map to the tag `http.1` unless it's override in the input. However, I added a mapping that can be set in `config.yaml` to map events to specific tags by event name as such:
```
monitoring:
  fluentbit:
    events:
      mapping:
      - name: abstraction.trigger
         tag: internal_api
```

### CODE REVIEW

File: pkg/types/event.go [hunk 1]
```diff
@@ -50,3 +51,12 @@ type EventWorkerLifecycleSchema struct {
 	WorkerID string `json:"worker_id"`
 	Status   string `json:"status"`
 }
+
+var EventAbstractionTriggerSchemaVersion = "1.0"
+
+type EventAbstractionTriggeredSchema struct {
```
Comment (luke-lombardi) [inline pkg/types/event.go:57]: Shouldn't this have an event_type field or something? this feels ambiguous

File: pkg/repository/base.go [hunk 2]
```diff
@@ -147,6 +147,7 @@ type EventRepository interface {
 	PushContainerStoppedEvent(containerID string, workerID string)
 	PushWorkerStartedEvent(workerID string)
 	PushWorkerStoppedEvent(workerID string)
+	PushAbstractionTriggeredEvent(workspaceId string, stub *types.Stub)
```
Comment (luke-lombardi) [inline pkg/repository/base.go:150]: I don't really like this name, it feels confusing. I'd almost argue for more specific functions even at the risk of more code / being more verbose. I./e. `PushDeployedStubEvent()` `PushServedStubEvent()`

File: pkg/repository/events.go [hunk 3]
```diff
@@ -25,9 +26,16 @@ func NewTCPEventClientRepo(config types.FluentBitEventConfig) EventRepository {
 		log.Println("[WARNING] fluentbit host does not appear to be up, events will be dropped")
 	}
 
+	// Parse event mapping
+	eventTagMap := make(map[string]string)
```
Comment (luke-lombardi) [inline pkg/repository/events.go:30]: you might not need the mapping if you use specific event types right?

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: pkg/types/event.go (hunk 1 above)
Comment (jsun-m) [reply pkg/types/event.go:57]: The stubtype includes the abstraction and event type

File: pkg/repository/events.go (hunk 3 above)
Comment (jsun-m) [reply pkg/repository/events.go:30]: The mapping is generic to easily map any event to different outputs

Comment (jsun-m) [reply pkg/repository/events.go:30]: That means users will be able to better direct where they want events to go. E.g. it goes to an API, elastic search, kinesis, or straight to slack through fluentbit

File: pkg/repository/base.go (hunk 2 above)
Comment (jsun-m) [reply pkg/repository/base.go]: I like this better. I renamed the subject from abstraction -> stub

**Review by luke-lombardi — APPROVED** [review]
Looks good. I'm still a little iffy on the event tag mapping but let's try it out.

**Pull request merged**